webrob/neems/manager.py

```diff
@@ -41,5 +46,7 @@ def filter_neems(self, query_string):
         #   - can an array be added to text index?
         #   - regex search possible, but only without index!!
         #       db.meta.find({"keywords": {"$regex": ".*robot.*","$options": 'i'}},{_id: 1}).pretty()
-        return map(lambda doc: doc["_id"], mongoDBMetaCollection.find(
-            {"$text": {"$search": query_string}}, {"_id": 1}))
+        neemHubSettings = get_settings(1)
+        if checkConnection(neemHubSettings) is not None:
+            return map(lambda doc: doc["_id"], checkConnection(neemHubSettings).find(
```

Comment:
this code is not optimal. each call of *checkConnection* establishes connection to mongo, so you would rather write something like that:

```
mongo = getMongo(neemHubSettings)
if mongo is not None:
    do_something_with(mongo)
```